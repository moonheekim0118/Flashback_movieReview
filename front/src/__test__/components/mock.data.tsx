export const MyInfoProp = 
{ 
    id:'test',
  nickname:'test',
  Reviews:1,
  profilePic:""
}  

export const MovieProp =  
{ 
    id:'test', 
    title:'test', 
    image:'test', 
    director:'test', 
    pubDate:'test'
};

export const AuthorProp = { id:'test', nickname:'test'};

export const ReviewProp = 
{ 
    id:'test', 
    Movie: MovieProp, 
    User:AuthorProp, 
    rating:'GOOD', 
    shortComment:'test',
    character:'test',
    line:'test',
    scene:'test',
    freeComment:'test',
};