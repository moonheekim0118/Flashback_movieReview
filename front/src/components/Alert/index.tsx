import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';


// 알림창 
const Alert=()=>{
    const { message, showAlert } = useSelector((state)=>state.alert);
    // showAlert 상태일때만 띄워준다. 

    return(
        <>
            {showAlert&&
            <Container>
                {message}
            </Container>}
        </>
    );
}

const Container = styled.div`
    position:fixed;
    left:50%;
    bottom:10%;
    transform:translateX(-50%);

    padding: 25px 15px;
    
    background-color:#cc00cc;
    border-raidus:10px;
    color:#fff;
    font-size:1.2rem;
    box-shadow:0px 0px 11px -1px rgba(0,0,0,0.75);

    z-index:7000;
`;

export default Alert;