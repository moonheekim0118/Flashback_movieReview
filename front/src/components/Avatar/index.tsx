import React from 'react';
import styled from 'styled-components';

interface Props {
    nickname?:string;
    imgSrc?:string;
    size?:number;
}

// 유저 아바타
const Avatar=({nickname, imgSrc="", size=40}:Props)=>{

    return(
        <Container size={size}>
            {imgSrc? <Image src={imgSrc}/> : <Nickname size={size}>{nickname[0]}</Nickname>}
        </Container>
    );
}

const Container = styled.div<{ size:number }>`
    width:${(props)=>`${props.size}px`};
    height:${(props)=>`${props.size}px`};
`;

const Image = styled.img`
    width:100%;
    height:100%;
    border-radius:100%;
    oveflow:hidden;
    object-fit:fill;
`;

const Nickname=styled.div<{ size:number }>`
    display:flex;
    justify-content:center;
    align-items:center;
    flex-direction:column;
    text-align:center;
    width:100%;
    height:100%;
    border-radius:100%;
    overflow:hidden;
    font-size:${(props)=>`${props.size-20}px`};
    color:#fff;
    background-color:#e0e0d1;
`;

export default Avatar;