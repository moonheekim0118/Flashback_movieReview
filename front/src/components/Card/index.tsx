import React from 'react';
import styled from 'styled-components';

interface Props {
  onClick?: (e: any) => void;
  posterSrc: string;
  alertChild?: React.ReactNode;
  descriptionChild: React.ReactNode;
  optionChild?: React.ReactNode;
}

const Card = ({
  onClick,
  posterSrc,
  alertChild,
  descriptionChild,
  optionChild,
}: Props) => {
  return (
    <Container onClick={onClick}>
      {alertChild}
      <MoviePoster src={posterSrc} alt="movie poster" />
      <MovieDescription>{descriptionChild}</MovieDescription>
      {optionChild}
    </Container>
  );
};

const Container = styled.article`
  display: flex;
  align-items: center;
  width: 100%;
  height: 200px;
  position: relative;

  border-bottom: 1px solid #f4f4f4;

  padding: 10px 20px;
  margin-top: 20px;
  cursor: pointer;
`;

const MoviePoster = styled.img`
  width: 150px;
  height: 100%;
  object-fit: scale-down;
  margin-right: 20px;
`;

const MovieDescription = styled.div`
  width: 100%;
  height: 50%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export default Card;
