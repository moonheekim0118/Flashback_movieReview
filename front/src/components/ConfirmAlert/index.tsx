import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

interface Props {
    text:string; // 알림창 메시지 
    clickYes:(e: any) => void; // 알림창에서 yes를 눌렀을 때 
    clickNo:(e: any) => void; // 알림차에서 No를 눌렀을 때 
}

// 확인 알림창
const ConfirmAlert=({text, clickYes, clickNo}:Props)=>{
    const root= document.getElementById('alert-root');
    
    return(
       ReactDOM.createPortal((
        <>
        <Overaly onClick={clickNo}/>
        <Container>
            <TextContainer>{text}</TextContainer>
            <ButtonContainer>
                <Button title="remove" id="submit" onClick={clickYes}>삭제</Button>
                <Button title="cancle" id="close" onClick={clickNo}>취소</Button>
            </ButtonContainer>
        </Container>
        </>
       ), root)
    );
}

const Overaly = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 8000;

  background-color: rgba(0, 0, 0, 0.3);
`


const Container = styled.div`
    display:flex;
    flex-direction:column;
    position:fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width:310px;
    height:150px;
    
    background-color:${props => props.theme.colors.bgColor};
    color:${props => props.theme.colors.fontColor};
    border-raidus:10px;
    border: 1px solid #e0e0d1;
    font-size:1.2rem;

    z-index:9000;
`;

const TextContainer = styled.div`
    width:100%;
    height:65%;
    padding: 30px 0;
    text-align:center;
    border-bottom: 1px solid #e0e0d1;
`;
const ButtonContainer = styled.div`
    height:35%;
    width:100%;
`;

const Button = styled.button<{ title: string }>`
    border: none;
    border-right:${(props)=>props.title==='remove' ? '1px solid #e0e0d1;' : 'none'};
    width:50%;
    background-color:inherit;
    padding: 17px 10px;
    color:${(props)=>props.title==='remove' ? 'red' : 'green'};
    font-weight:bold;
    font-size:1rem;

    cursor:pointer;

    &:hover{
        background-color:#e0e0d1;
    }
`;

export default ConfirmAlert;