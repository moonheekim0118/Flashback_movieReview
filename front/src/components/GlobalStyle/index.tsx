import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    *,
    *::after,
    *::before{
        box-sizing:border-box;
        margin:0;
        padding:0;
        font-family: 'Nanum Myeongjo', serif;
    }
    *:focus{
        outline:none;
    }
    body{
        overflow-x:hidden;
        color:#000000;
        background-color:#FFF;
    }
`;

export default GlobalStyle;
