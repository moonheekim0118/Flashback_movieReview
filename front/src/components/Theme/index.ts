export const lightTheme= {
   colors:{
       bgColor:'#fff',
       fontColor:'black'
   }
}


export const darkTheme= {
  colors:{
      bgColor:'#1a1a1a',
      fontColor:'#fff'
  }
}