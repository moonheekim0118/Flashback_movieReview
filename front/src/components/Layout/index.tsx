import React ,{ useCallback , useEffect , useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Link from 'next/link';
import Router from 'next/router'
import Alert from '../../components/Alert';
import Icons from '../../atoms/Icons';
import styled from 'styled-components';
import { CLOSE_ALERT } from '../../actions/alert';
import { faChevronLeft, faBars } from '@fortawesome/free-solid-svg-icons';
import Logout from '../User/Logout';
import { clear } from 'console';

interface Props {
    PageName?:string;
    children?:any;
}

const Layout=({ PageName="" , children} : Props)=>{
    const dispatch = useDispatch();
    const { showAlert } = useSelector((state)=>state.alert);
    const loginDone = useSelector(state=>state.user.loginDone);
  
    const onPushBack = useCallback(()=>{
        Router.back();
    },[]);

    useEffect(()=>{
        if(showAlert){
            const timer = setTimeout(()=>dispatch({type:CLOSE_ALERT}),5000);
            return()=>clearTimeout(timer);
        }
    },[showAlert]);

    const MainButton = PageName ==='메뉴' && loginDone ? <Logout/> : 
    <Link href="/menu"><a><Icons icon={faBars} color="lightPurple"/></a></Link> ;
    
    return(
        <App>
        <Alert/>
        {PageName &&         
        <Header>
            <div><Icons icon={faChevronLeft} className={"fa-chevron-left"} onClick={onPushBack}/></div>
            <PageInfo>
                <PageDescription>{PageName}</PageDescription>
                {MainButton}
            </PageInfo>
        </Header>}
        {children}
        </App>
    );
}

const App = styled.div`
    min-height: 100vh;
    width:767px;
    margin:auto;
    position:relative;
    
    background-color:${props => props.theme.colors.bgColor};
    color:${props => props.theme.colors.fontColor};

    @media screen and (max-width:414px){
        width: 100%;
    };

    @media screen and (max-width:768px){
        width: 375px;
    };

`;
const Header = styled.header`
    display:flex;
    justify-content:space-between;
    align-items:center;

    width: 100%;
    height: 50px;
    padding:0px 20px;

    position:sticky;
    top: 0;  
    
    color:#cc00cc;
    border-bottom:1px solid #e6b3cc;
    background-color:inherit;

    z-index:2000;
    
`;

const PageInfo = styled.div`
    width:200px;
    max-width:80%;
    display:flex;
    justify-content:space-evenly;
    align-items:center;
`;

const PageDescription = styled.div`
    padding-left:20px;
`;

export default Layout;