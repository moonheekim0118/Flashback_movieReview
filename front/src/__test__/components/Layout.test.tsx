import 'jsdom-global/register';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../components/Theme';
import { useSelector } from 'react-redux';
import { mount } from 'enzyme';
import Layout from '../../components/Layout';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(() => jest.fn()),
}));

describe('<Layout/>', () => {
  let container;
  beforeEach(() => {
    const state = { showAlert: true, loginDone: true };
    useSelector.mockImplementation(() => state);
    container = mount(
      <ThemeProvider theme={lightTheme}>
        <Layout PageName="메뉴" />
      </ThemeProvider>
    );
  });

  it('should match correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
