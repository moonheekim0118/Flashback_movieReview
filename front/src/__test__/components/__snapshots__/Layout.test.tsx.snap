// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> should match correctly 1`] = `"<div class=\\"Layout__App-sc-70i2gn-0 bdPUDV\\"><div class=\\"Alert__Container-l9ph2f-0 lwAmd\\"></div><header class=\\"Layout__Header-sc-70i2gn-1 hzSsnj\\"><div><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"chevron-left\\" class=\\"svg-inline--fa fa-chevron-left fa-w-10 Icons__StyledIcon-khcp93-0 bOobqK faChevronLeft\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\" iconsize=\\"24\\" color=\\"purple\\"><path fill=\\"currentColor\\" d=\\"M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z\\"></path></svg></div><div class=\\"Layout__PageInfo-sc-70i2gn-2 gxWtzz\\"><div class=\\"Layout__PageDescription-sc-70i2gn-3 iFkwIu\\">메뉴</div><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"sign-out-alt\\" class=\\"svg-inline--fa fa-sign-out-alt fa-w-16 Icons__StyledIcon-khcp93-0 jWqSFQ faSignOutAlt\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" iconsize=\\"24\\" color=\\"lightPurple\\"><path fill=\\"currentColor\\" d=\\"M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z\\"></path></svg></div></header></div>"`;
